docs/kubernetes/kctlr-create-service-vs.rst

```diff
@@ -0,0 +1,137 @@
+Define a virtual server for a Service
+-------------------------------------
+
+#. Define the virtual server you want to create in an :ref:`F5 resource JSON blob <f5-resource-blob>`.
+#. Include the JSON blob in the :code:`data` section of a Kubernetes `ConfigMap`_ resource.
+
+.. _kctlr configmap example:
+
+HTTP example
+````````````
+
+For example, if your Service looks like this:
+
+.. code-block:: yaml
+
+   apiVersion: v1
+   kind: Service
+   metadata:
+     name: mySuperAwesomeApp
+     labels:
+       app: myApp
+   spec:
+     ports:
+     - protocol: TCP
+       port: 80
+       targetPort: 9376
+     type: clusterIP
+
+Your F5 resource ConfigMap might look like this:
+
+.. literalinclude:: /kubernetes/config_examples/f5-resource-vs-example.configmap.yaml
+   :lines: 1-7,9-15,23-34
+
+:fonticon:`fa fa-download` :download:`f5-resource-vs-example.configmap.yaml </kubernetes/config_examples/f5-resource-vs-example.configmap.yaml>`
+
+.. seealso::
+   :class: sidebar
+
+   See :ref:`kctlr-manage-bigip-objects` for information about :ref:`adding health monitors <k8s-service healthmonitor>`, :ref:`using IPAM to assign virtual server IP addresses <kctlr-ipam>`, and more.
+
+- The :code:`backend.servicePort` property in the F5 resource matches the :code:`ports.port` property in the Service definition. The |kctlr| uses this to relate the Pod Node Ports and Endpoints to the BIG-IP virtual server.
+
+- The :code:`ports.targetPort` setting is the Pod/Container port to which you want to send traffic.
+
+- You can replace :code:`balance: round-robin` with any supported BIG-IP load balancing mode. [#lb]_
+
+HTTPS example
+`````````````
+
+For an HTTPS virtual server, your ConfigMap for the same Service might look like this:
+
+.. literalinclude:: /kubernetes/config_examples/f5-resource-configmap-https.yaml
+   :lines: 1-14,22-36
+
+:fonticon:`fa fa-download` :download:`f5-resource-configmap-https.yaml </kubernetes/config_examples/f5-resource-configmap-https.yaml>`
+
+- You can define :code:`sslProfile.f5ProfileName` using any existing BIG-IP client SSL profile.
+
+- To provide a list of SSL profiles, use :code:`sslProfile.f5ProfileNames`.
+
+.. _upload vs ConfigMap:
+
+Upload the ConfigMap to the API Server
+--------------------------------------
+
+.. tip::
+
+   If you want to create both HTTP and HTTPS virtual servers (in other words, expose multiple ports) for the same Service, create an F5 resource ConfigMap for each port.
+   You can pass the names of both YAML files in your :code:`apply` command or include both resources in a single manifest file.
+
+
+Kubernetes
+``````````
+
+.. include:: /_static/reuse/kubectl-apply.rst
+
+OpenShift
+`````````
+
+.. include:: /_static/reuse/oc-apply.rst
+
+.. _verify bigip service objects created:
+
+Verify changes on the BIG-IP system
+-----------------------------------
+
+.. include:: /_static/reuse/verify-bigip-objects.rst
+
+
+.. _kctlr-downed-services:
+
+Virtual Servers for down or replaced Services
```

Comment:
We can't remove a Service nor a ConfigMap without removing the virtual server and BIG-IP objects. The only way you can leave BIG-IP untouched would be to remove the ctlr...run tests, maintenance, changes...then restart the controller. Whatever changes would be overwritten by the ctlr on restart though unless care was taken to reflect those changes via kubernetes.